modules/system/classes/CompilableAssets.php

```diff
@@ -0,0 +1,340 @@
+<?php namespace System\Classes;
+
+use Cms\Classes\Theme;
+use System\Classes\PluginManager;
+use Winter\Storm\Exception\SystemException;
+use Winter\Storm\Filesystem\PathResolver;
+use Winter\Storm\Support\Facades\Config;
+use Winter\Storm\Support\Facades\File;
+use Winter\Storm\Support\Str;
+
+/**
+ * Compilable assets using for Node.js compilation and processing.
+ *
+ * This works similar to the `System\Classes\CombineAssets` class in that it allows modules, plugins and themes to
+ * register configurations that will be passed on to Node.js based compilers for processing.
+ *
+ * @package winter\wn-system-module
+ * @author Jack Wilkinson <me@jackwilky.com>
+ * @copyright Winter CMS Maintainers
+ */
+class CompilableAssets
+{
+    use \Winter\Storm\Support\Traits\Singleton;
+
+    /**
+     * The filename that stores the package definition.
+     */
+    protected string $packageJson = 'package.json';
+
+    /**
+     * @var array<string, array<string, string>> List of package types and registration methods
+     */
+    protected array $compilableConfigs = [
+        'mix' => [
+            'registrationMethod' => 'registerMixPackages',
+            'configFile' => 'winter.mix.js'
+        ],
+        'vite' => [
+            'registrationMethod' => 'registerVitePackages',
```

Comment:
Does this need to be a config item? Can't it be inferred from the name of the array? Does it support some level of extensibility this way that I'm not seeing at the moment?